corehq/apps/celery/__init__.py

```diff
@@ -18,3 +18,25 @@ def _init_celery_app():
     app.config_from_object('django.conf:settings', namespace='CELERY')
     app.autodiscover_tasks()
     app.set_default()
+
+
+class PeriodicTask(Task):
+
+    @classmethod
+    def on_bound(cls, app):
+        app.conf.beat_schedule[cls.name] = {
+            'task': cls.name,
+            'schedule': cls.run_every,
+            'args': (),
+            'kwargs': {},
+            'options': cls.options or {}
+        }
+
+
+def periodic_task(*args, **options):
+    if options.get('queue'):
+        options['options'] = {
+            'queue': options.pop('queue')
+        }
+    options['base'] = PeriodicTask
+    return app.task(**options)
```

Comment:
I think these should live in a different module. Suggestion: `corehq.apps.celery.periodic`. And `periodic_task` should be imported in this module (`__init__.py`) to make it easier to find in other places.

Reply:
https://github.com/dimagi/commcare-hq/pull/31584/commits/28ea3e2ebfdb4c80801778ef53f95b930e8be9b8  https://github.com/dimagi/commcare-hq/pull/31584/commits/d0187705527d0e5c123facedfef63472c52e290e